eng/common/TestResources/New-TestResources.ps1

```diff
@@ -106,14 +119,27 @@ param (
 
 . $PSScriptRoot/SubConfig-Helpers.ps1
 
-$azsdkPipelineVnet = "/subscriptions/a18897a6-7e44-457d-9260-f2854c0aca42/resourceGroups/azsdk-pools/providers/Microsoft.Network/virtualNetworks/azsdk-pipeline-vnet-wus"
-$azsdkPipelineSubnets = @(
-    ($azsdkPipelineVnet + "/subnets/pipeline-subnet-ubuntu-1804-general"),
-    ($azsdkPipelineVnet + "/subnets/pipeline-subnet-ubuntu-2004-general"),
-    ($azsdkPipelineVnet + "/subnets/pipeline-subnet-ubuntu-2204-general"),
-    ($azsdkPipelineVnet + "/subnets/pipeline-subnet-win-2019-general"),
-    ($azsdkPipelineVnet + "/subnets/pipeline-subnet-win-2022-general")
-)
+$azsdkPipelineVnetWestUS = '/subscriptions/a18897a6-7e44-457d-9260-f2854c0aca42/resourceGroups/azsdk-pools/providers/Microsoft.Network/virtualNetworks/azsdk-pipeline-vnet-wus'
+$azsdkPipelineVnetCanadaCentral = '/subscriptions/a18897a6-7e44-457d-9260-f2854c0aca42/resourceGroups/azsdk-pools/providers/Microsoft.Network/virtualNetworks/azsdk-pipeline-vnet-cnc'
+$azsdkPipelineSubnetMap = @{
+    'azsdk-pool-mms-ubuntu-1804-general' = ($azsdkPipelineVnetWestUS + '/subnets/pipeline-subnet-ubuntu-1804-general')
+    'azsdk-pool-mms-ubuntu-2004-general' = ($azsdkPipelineVnetWestUS + '/subnets/pipeline-subnet-ubuntu-2004-general')
```

Comment:
Explicitly mapping here means any new pool will also need to update here. Should we do it via string replacement instead so that new pools could work without a change if they follow the correct convention?

Reply:
The challenge here is that the vnets have to be region paired with the pools. If we wanted to do a convention based naming scheme, I think we would also want to change our pool names to include the region they are in so we can know what vnet to pair them with. And we currently use the 1es change region feature (which changes the VMSS region despite the pool region) to make things more confusing.

Part of me doesn't like the config being stored in multiple places (especially here), and the other part is worried about the general practice of naming magic making things hard to diagnose. Open to ideas here.

Another option is we could make a call to Azure at the time of running this script to get the pool/subnet info. But that involves a few more pools that we actually use here and could add some time to the pipeline.

Reply:
We also will need to come up with some scripts to auto-update all our resources. As it is right now, we have many places across our services (apiview, etc.) that have these same subnets added. Whether it is a manual or automated update there is still downtime potential (e.g. we found you have to restart app service after adding a vnet+subnet among other things)

Reply:
My main concern is that I would prefer we not be forced to update this table anytime we add a new pool. Is there a property we can query on the pool itself? If so, maybe that call is worth having here.

Reply:
Looks like we can do it like so, and it runs very quickly:

```
(Get-AzResource -ResourceGroupName azsdk-pools -Name azsdk-pool-mms-ubuntu-2004-general -ExpandProperties).Properties.networkProfile
```

Reply:
And our provisioner app should already have read access here, as we need a certain level of permissions to able to link these subnets to other resources.

Reply:
Testing out the following:

```
    $ctx = Get-AzContext
    Set-AzContext -Subscription 'Azure SDK Engineering System'
    $poolSubnet = (Get-AzResource -ResourceGroupName azsdk-pools -Name $env:Pool -ExpandProperties).Properties.networkProfile.subnetId
    $ctx | Set-AzContext
```

Reply:
Lets do that then.